src/mail/editor/SendMailModel.ts

```diff
@@ -878,55 +885,78 @@ export class SendMailModel {
 		return !isSecurePassword(minimalPasswordStrength)
 	}
 
+	saveDraft(
+		saveAttachments: boolean,
+		mailMethod: MailMethod,
+		blockingWaitHandler: (arg0: TranslationKey | lazy<string>, arg1: Promise<any>) => Promise<any>,
+	): Promise<void> {
+
+		if (this.currentSavePromise == null) {
+			this.currentSavePromise = this.doSaveDraft(saveAttachments, mailMethod).then(async () => {
+				this.currentSavePromise = null
```

Comment:
Also this is dangerous as it's only on fulfillment, would we try again after an error or are we stuck?

Reply:
I guess on error then the chain would get cancelled? But the state would need to be reset too. I would write a test for this case as well.